apps/studio/components/interfaces/APIKeys/ApiKeyPill.tsx

```diff
@@ -0,0 +1,195 @@
+      cacheTime: 0, // Don't cache the key data
+    }
+  )
+
+  // Auto-hide timer for the API key (security feature)
+  useEffect(() => {
+    if (show && data?.api_key) {
+      // Auto-hide the key after 10 seconds
+      const timer = setTimeout(() => {
+        setShowState(false)
+        // Clear the cached key from memory
+        queryClient.removeQueries({
+          queryKey: apiKeysKeys.single(projectRef, apiKey.id as string),
+          exact: true,
+        })
+      }, 10000) // Hide after 10 seconds
+
+      return () => clearTimeout(timer)
+    }
+  }, [show, data?.api_key, projectRef, queryClient, apiKey.id])
+
+  async function onSubmitShow() {
+    // Don't reveal key if not allowed or loading
+    if (isSecret && !canManageSecretKeys) return
+    if (isLoadingPermission) return
+
+    // This will enable the API key query to fetch and reveal the key
+    setShowState(true)
+  }
+
+  async function onCopy() {
+    // If key is already revealed, use that value
+    if (data?.api_key) return data?.api_key
+
+    try {
+      // Fetch full key and immediately clear from cache after copying
+      const result = await refetchApiKey()
+      queryClient.removeQueries({
+        queryKey: apiKeysKeys.single(projectRef, apiKey.id as string),
+        exact: true,
+      })
+
+      if (result.isSuccess) return result.data.api_key
+
+      if (error) {
+        toast.error('Failed to copy secret API key')
+      }
+    } catch (error) {
+      console.error('Failed to fetch API key:', error)
+    }
+
+    // Fallback to the masked version if fetch fails
+    return apiKey.api_key
+  }
+
+  // States for disabling buttons/showing tooltips
+  const isRestricted = isSecret && !canManageSecretKeys
+  const isLoading = isLoadingPermission
+
+  return (
+    <>
+      <AnimatePresence mode="wait" initial={false}>
+        <div
+          className={cn(
+            InputVariants({ size: 'tiny' }),
+            'flex-1 grow gap-0 font-mono rounded-full overflow-hidden',
+            show ? 'ring-1 ring-foreground-lighter ring-opacity-50' : 'ring-0 ring-opacity-0',
+            'transition-all',
+            'max-w-[340px]',
+            'cursor-text',
+            'relative'
+          )}
+          style={{ userSelect: 'all' }}
+        >
+          <span>{apiKey?.api_key.slice(0, 15)}</span>
+          <motion.span
+            key={show ? 'shown' : 'hidden'}
+            initial={{ opacity: 0, y: show ? 16 : -16 }}
+            animate={{ opacity: 1, y: 0 }}
+            exit={{ opacity: 0, y: show ? 16 : -16 }}
+            transition={{
+              duration: 0.1,
+              y: { type: 'spring', stiffness: 1000, damping: 55 },
+            }}
+            className="truncate"
+          >
+            {show && data?.api_key ? data?.api_key.slice(15) : '••••••••••••••••'}
+          </motion.span>
+        </div>
+      </AnimatePresence>
+
+      {/* Reveal button - only shown for secret keys and when not already revealed */}
```

Comment:
Feels like this should always stay? 
it currently disappears when you click it, but feels like it should toggle the open eye / closed eye icon instead?

Reply:
> Feels like this should always stay? it currently disappears when you click it, but feels like it should toggle the open eye / closed eye icon instead?

I liked that the copy button moved into place after, which is inevitably what you might want to do.